src/runtime-rs/crates/hypervisor/src/qemu/cmdline_generator.rs

```diff
@@ -1686,6 +1897,106 @@ impl<'a> QemuCmdLine<'a> {
         self.devices.push(Box::new(console_socket_chardev));
     }
 
+    pub fn add_pcie_root_port(
+        &mut self,
+        total_ports: u32,
+        machine_type: &str,
+        memsz_32bit: u32,
+        memsz_64bit: u64,
+    ) -> Result<()> {
+        let (bus, chassis, multi_function, addr) = match machine_type {
+            "QemuQ35" | "QemuVirt" => (
+                DEFAULT_BRIDGE_BUS.to_string(),
+                "0".to_string(),
+                false,
+                "0".to_string(),
+            ),
+            _ => return Ok(()),
+        };
+
+        for index in 0..total_ports {
+            self.devices.push(Box::new(PCIeRootPortDevice {
+                id: format!("{}{}", PCIePortBusPrefix::RootPort, index),
+                bus: bus.clone(),
+                chassis: chassis.clone(),
+                slot: index.to_string(),
+                multifunction: multi_function,
+                addr: addr.clone(),
+                io_reserve: String::new(),
+                bus_reserve: String::new(),
+                pref32_reserve: String::new(),
+                pref64_reserve: format!("{}B", memsz_64bit),
+                mem_reserve: format!("{}B", memsz_32bit),
+            }));
+        }
+
+        Ok(())
+    }
+
+    pub fn add_pcie_switch_port(
```

Comment:
Could we have a comment as to what this function does, or maybe at least import the [diagram from runtime-go](https://github.com/kata-containers/kata-containers/blob/378f454fb98463f9ecd3afd9e9c680b2c708be39/src/runtime/virtcontainers/qemu.go#L2578) since if I'm not mistaken, that's the structure this function builds?